gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,611 @@
+    Mutation updatedMutation =
+        mutation.toBuilder().setNewLocation(Optional.ofNullable(newPoint)).build();
+
+    localDataStore
+        .updateMutations(ImmutableList.<Mutation>builder().add(updatedMutation).build())
+        .test()
+        .assertComplete();
+
+    ImmutableList<Mutation> savedMutations =
+        localDataStore.getPendingMutations(updatedMutation.getFeatureId()).blockingGet();
+    Assert.assertEquals(1, savedMutations.size());
+
+    FeatureMutation savedMutation = (FeatureMutation) savedMutations.get(0);
+    Assert.assertEquals(newPoint, savedMutation.getNewLocation().get());
+  }
+
+  @Test
+  public void testRemovePendingMutation() {
+    User user = createTestUser();
+    localDataStore.insertOrUpdateUser(user).test().assertComplete();
+
+    Project project = createTestProject();
+    localDataStore.insertOrUpdateProject(project).test().assertComplete();
+
+    Layer layer = project.getLayers().get(0);
+    FeatureMutation mutation =
+        createTestFeatureMutation(user.getId(), project.getId(), layer.getId());
+    localDataStore.applyAndEnqueue(mutation).test().assertComplete();
+
+    localDataStore
+        .removePendingMutations(ImmutableList.<Mutation>builder().add(mutation).build())
+        .test()
+        .assertComplete();
+
+    localDataStore
+        .getPendingMutations(mutation.getFeatureId())
+        .test()
+        .assertValue(AbstractCollection::isEmpty);
+  }
+
+  @Test
+  public void testMergeFeature() {
+    User user = createTestUser();
+    localDataStore.insertOrUpdateUser(user).test().assertComplete();
+
+    Project project = createTestProject();
+    localDataStore.insertOrUpdateProject(project).test().assertComplete();
+
+    Layer layer = project.getLayers().get(0);
+    FeatureMutation mutation =
+        createTestFeatureMutation(user.getId(), project.getId(), layer.getId());
+    localDataStore.applyAndEnqueue(mutation).test().assertComplete();
+
+    Feature feature = localDataStore.getFeature(project, mutation.getFeatureId()).blockingGet();
+    Point point = Point.newBuilder().setLongitude(11.0).setLatitude(33.0).build();
+    feature = feature.toBuilder().setPoint(point).build();
+    localDataStore.mergeFeature(feature).test().assertComplete();
+
+    Feature newFeature = localDataStore.getFeature(project, mutation.getFeatureId()).blockingGet();
+    Assert.assertEquals(mutation.getFeatureId(), newFeature.getId());
+    Assert.assertEquals(project, newFeature.getProject());
+    Assert.assertEquals(layer.getItemLabel(), newFeature.getTitle());
+    Assert.assertEquals(layer, newFeature.getLayer());
+    Assert.assertNull(newFeature.getCustomId());
+    Assert.assertNull(newFeature.getCaption());
+    Assert.assertEquals(point, newFeature.getPoint());
+    Assert.assertEquals(user, newFeature.getCreated().getUser());
+    Assert.assertEquals(user, newFeature.getLastModified().getUser());
+  }
+
+  @Test
+  public void testApplyAndEnqueue_observationMutation() {
```

Comment:
This test (and ones like it) look they're testing more than one behavior, which is why they're longer . Can we split these into clear preconditions, execute, and postconditions? See previous comments for ways to simplify.

Reply:
Done. PTAL